Tensile/Hardware.py

```diff
@@ -33,13 +33,20 @@ def FromISA(cls, isa):
         return cls("AMDGPU", value=cls("Processor", value=gfxArch))
 
     @classmethod
-    def FromHardware(cls, isa, cuCount=None):
+    def FromHardware(cls, isa, cuCount=None, isAPU=None):
         gfxArch = Common.gfxName(isa)
-        if cuCount == None:
+        if cuCount == None and isAPU == None:
```

Comment:
If neither CUCount or IsAPU is set, the the library generating code expects the architecture to just be a "Processor" predicate rather than a dictionary. Could be cleaned up a bit, but it'd require changes across a few other files as well. This might be a good candidate to clean up in the future, but I'd rather leave as is for this PR to unblock tuning work. I'll file a ticket for this one.